mesheryctl/pkg/utils/auth.go

```diff
@@ -372,6 +413,26 @@ func selectProviderPrompt(provs map[string]Provider) Provider {
 	}
 }
 
+func chooseProviderDirect(provs map[string]Provider, option string) (Provider, error) {
+	provArray := []Provider{}
+	provNames := []string{}
+
+	for _, prov := range provs {
+		provArray = append(provArray, prov)
+	}
+
+	for _, prov := range provArray {
+		provNames = append(provNames, prov.ProviderName)
+	}
+
+	for i := 0; i < len(provNames); i++ {
+		if provNames[i] == option {
+			return provArray[i], nil
+		}
+	}
+	return provArray[1], fmt.Errorf("the specified provider '%s' is not available. Please try giving correct provider name", option)
```

Comment:
why don't we directly iterate the map itself?? and match _lowercased_ keys with the _lowercased_ option

Reply:
Never thought at that time 😶

Reply:
Function refactored

Reply:
I don't think the function is optimised, we're increasing the memory use here, isn't iterating the map itself a good approach in terms of code?

Reply:
Will change soon